flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicSource.java

```diff
@@ -179,13 +183,37 @@ public ScanRuntimeProvider getScanRuntimeProvider(ScanContext context) {
 	@Override
 	public Map<String, DataType> listReadableMetadata() {
 		final Map<String, DataType> metadataMap = new LinkedHashMap<>();
+
+		// add value format metadata with prefix
```

Comment:
Add a comment to mention why we must put the format metadata before connector metadata? I think the order can't be switched.

Reply:
actually the order doesn't matter here, but I just noticed that the order should be the other way around if format metadata should have higher precedence